addons/crm/tests/test_crm_lead.py

```diff
@@ -178,3 +178,37 @@ def test_crm_lead_merge(self):
         self.assertEqual(merged_opportunity.partner_id.id, self.env.ref("base.res_partner_3").id, 'Partner mismatch')
         self.assertEqual(merged_opportunity.type, 'opportunity', 'Type mismatch: when opps get merged together, the result should be a new opp (instead of %s)' % merged_opportunity.type)
         self.assertFalse(test_crm_opp_03.exists(), 'This tailing opp (id %s) should not exist anymore' % test_crm_opp_03.id)
+
+    def test_lead_won(self):
+        """
```

Comment:
Not sure why you create 3 identical leads for each stage?

Reply:
In fact, this test was mainly to check if each lead , after winning them, had the correct final stage id. But as we won't use this final_stage_id, this test can be simplified on only one 1 record.